notifications_utils/recipients.py

```diff
@@ -263,11 +298,31 @@ def column_headers_as_column_keys(self):
 
     @property
     def missing_column_headers(self):
-        return set(
-            key
-            for key in self.placeholders
-            if (Columns.make_key(key) not in self.column_headers_as_column_keys and not self.is_optional_address_column(key))
-        )
+        """
+        Missing headers must return the following in each case:
+
+        Eg 1: file: []
+        placeholders: [phone number, name]
+        missing: [phone number, name]
+
+        Eg 2: file: [adresse courriel]
+        placeholders: [email address]
+        missing:  []
+
+        """
+        result = set()
+        for key in self.placeholders:
+            # If the key has a different heading due to language, then we need to check that the column_headers
+            # in the file have either the same or other language. This is only for email address / phone number
+            if key in self.recipient_column_headers_lang_check:  # type: ignore
```

Comment:
So do we allow FR column headers when the request is in EN and vice versa?  What exactly is the intersection method doing here?

Reply:
yes we allow FR heading and EN request and vice versa.

The intersection says - if the headings that english/french are not in column headers of the file:
Add that header to the result - we know that it is "missing"